url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/inport/StaticImport.java

```diff
@@ -0,0 +1,2125 @@
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Pants s : pants) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Pants;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Gloves s : gloves) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Gloves;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Armor s : armors) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Armor;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Helm s : helms) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Helm;" + s.getItemset().getName()
+					+ ";" + s.getName() + ";" + s.getLevel() + ";"
+					+ s.getStars() + ";" + s.getItemset().getUsefulChars()
+					+ ";" + s.getTyp());
+			sb.append("\n");
+		}
+
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream("data"
+							+ File.separator + "prices.csv"), "UTF-8"));
+
+			out.write(sb.toString());
+			out.close();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private static void importPrices() {
+		List<String> slist = new ArrayList<>();
+		try {
+			BufferedReader br = new BufferedReader(
+					new InputStreamReader(new FileInputStream("data"
+							+ File.separator + "prices.csv"), "UTF-8"));
+
+			String line = br.readLine();
+			slist.add(line);
+			while (line != null) {
+				line = br.readLine();
+				slist.add(line);
+			}
+			br.close();
+		} catch (Exception e) {
+		}
+
+		for (Shoes s : shoes) {
+			itemmap.put(s.getId(), s);
+		}
+		for (Pants s : pants) {
+			itemmap.put(s.getId(), s);
+		}
+		for (Gloves s : gloves) {
+			itemmap.put(s.getId(), s);
+		}
+		for (Armor s : armors) {
+			itemmap.put(s.getId(), s);
+		}
+		for (Helm s : helms) {
+			itemmap.put(s.getId(), s);
+		}
+
+		for (int i = 1; i < slist.size() - 1; i++) {
+			String x = slist.get(i);
+			String[] split = x.split(";");
+			int id = new Integer(split[0]);
+			double newprice = new Double(split[1].replace(",", "."));
+
+			Item itam = itemmap.get(id);
+
+			if (itam != null) {
+				itam.setPrice(newprice);
+			}
+
+		}
+
+	}
+
+	private static void exportItems() {
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 2090, 2091, 2097. The resource is closed at line 2106. There are other execution paths that do not contain closure statements, for example, when Marshaller.marshal throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).